### DISCUSSION TITLE
the gpu-instantiated components will disappear when I rotate the model in cesium

### DISCUSSION THREAD

**Discussion by varLiu** [zeux/meshoptimizer#565] [Q&A]
I modified gltfpack to preserve the other extensions, but the gpu-instantiated components will disappear when I rotate the model in cesium.
I asked the same question in the cesium community.
[https://community.cesium.com/t/the-gpu-instantiated-artifacts-will-disappear-when-i-rotate-the-model-in-cesium/24688?u=varliu](url)
I also hope you could help me see if there were any errors at the data of gltf.
the gltf of my 3dtiles support the following extensions:
EXT_structural_metadata,
EXT_mesh_features,
EXT_instance_features,
EXT_mesh_gpu_instancing,
EXT_meshopt_compression(gltfpack),
KHR_mesh_quantization(gltfpack),
KHR_texture_basisu(gltfpack),
KHR_texture_transform(gltfpack)

1.It has not been processed by gltfpack.
![model_no_gltfpack](https://github.com/zeux/meshoptimizer/assets/51195718/c8866b13-98f7-4a8f-8f05-e05e8b82cf82)
[model_no_gltfpack.zip](https://community.cesium.com/uploads/short-url/3klWcVbLxGWeeEruWgaEYsF88D3.zip) (29.8 MB)

2.It has been processed by gltfpack.
![model_gltfpack](https://github.com/zeux/meshoptimizer/assets/51195718/a7208a0d-4ee9-4045-bd9d-c82ad7917f87)
[model_gltfpack.zip](https://community.cesium.com/uploads/short-url/zQKze1VMJBJWLrhcDsinrSarDdd.zip) (5.3 MB)

**Comment by zeux** [comment]
I think you might need to upload the files to GitHub because the links presumably are only accessible with a Cesium discourse account.

**Reply by varLiu** [reply]
https://github.com/zeux/meshoptimizer/discussions/565#discussioncomment-6147488

**Reply by zeux** [reply]
Thanks! I looked at a couple of tiles in isolation but they looked the same in gltf-viewer before and after. Do you happen to know which tile contains the problematic objects? For me to debug this I'd like to be able to just look at one specific glTF file in a viewer instead of setting up Cesium viewer.

**Reply by varLiu** [reply]
After many debugs, I found that this phenomenon occurs in cesium only if the KHR_mesh_quantization is enabled.
My 3dtiles doesn't have any problems now if the KHR_mesh_quantization isn't enabled.
Maybe it's a Cesium bug?

**Comment by varLiu** [comment]
oh sorry.
Here you are.

1.It has not been processed by gltfpack.
[https://github.com/varLiu/meshoptimizer/blob/feat-3dtiles/issue_models/model_no_gltfpack.zip](url)

2.It has been processed by gltfpack.
[https://github.com/varLiu/meshoptimizer/blob/feat-3dtiles/issue_models/model_gltfpack.zip](url)
